pkg/perf/monitor.go

```diff
@@ -0,0 +1,75 @@
+package perf
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-co-op/gocron"
+	"github.com/gomodule/redigo/redis"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zos/pkg/utils"
+)
+
+// PerformanceMonitor holds the module data
+type PerformanceMonitor struct {
+	scheduler *gocron.Scheduler
+	redisConn redis.Conn
+	tasks     []Task
+}
+
+// NewPerformanceMonitor returns PerformanceMonitor instance
+func NewPerformanceMonitor(redisAddr string) (*PerformanceMonitor, error) {
+	redisPool, err := utils.NewRedisPool(redisAddr)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed creating new redis pool")
+	}
+
+	redisConn := redisPool.Get()
```

Comment:
This is not how u use the redis Pool. The idea of a Pool of connection that u Get a connection from the pool, use it and then close it immediately when you are done using it. 

Closing a connection will return the connection to the pool (the pool might not close the connection) but keep it alive for the next one to need a connection (by calling the pool.Get()) 

So what u are doing is breaking the pooling contract by hording a connection forever. This is also really bad because if the connection is interrupted (or lost because of any reason) there is NO way that u renew the connection. The pool on the other hand tests the connection before it give u one when u do a Get, and renew it if needed.

In summary, what u need to do is to keep the pool on you structure (not a connection) and in you setCache and Get methods what u need to do is

```go
con := pm.pool.Get()
defer con.Clonse()

// use the con to get or set
```